src/internal/m365/graph/errors.go

```diff
@@ -128,11 +132,26 @@ func IsErrQuotaExceeded(err error) bool {
 }
 
 func IsErrExchangeMailFolderNotFound(err error) bool {
-	return hasErrorCode(err, resourceNotFound, mailboxNotEnabledForRESTAPI)
+	return hasErrorCode(err, resourceNotFound, itemNotFound, mailboxNotEnabledForRESTAPI)
 }
 
 func IsErrUserNotFound(err error) bool {
-	return hasErrorCode(err, requestResourceNotFound)
+	if hasErrorCode(err, requestResourceNotFound, invalidUser) {
+		return true
+	}
+
+	if hasErrorCode(err, resourceNotFound) {
+		odErr, ok := err.(odataerrors.ODataErrorable)
+		if !ok {
+			return false
+		}
```

Comment:
not really sure what the best strategy here is, but other errors are also checking for `resourceNotFound` so not trying to check what exactly wasn't found could lead to false-positives

Reply:
I think this handling is fine, but shouldn't we also now check for the exclusion in those other comparisons?  IE: certain error checks and compare just the code, while others (like this) need to compare both the code and message, and any case that checks both must always check both?

Reply:
I'll follow up on this in another PR. There's probably a few things that should be adjusted beyond this